### PR TITLE
Fixes #38019 - Add hammer support for flatpak remotes and remote repositories

### PR DESCRIPTION
**Pull request by sjha4**
Adds hammer support for flatpak remotes. The sub-commands added are list/show/create/update/delete and scan.
This also supports list/info on remote repository and adds a command to mirror remote repository into katello.

This requires https://github.com/Katello/katello/pull/11223 for testing..

Working on more tests but the commands should be ready for testing.

### CODE REVIEW

**Comment by sjha4** [comment]
Test failures are due to rubocop autocorrect on my last push..Will push..

**Review by ianballou — COMMENTED** [review]
It seems there are missing organization options for deleting remotes:

```
Usage:
    hammer flatpak-remote <delete|destroy> [OPTIONS]

Options:
 --id NUMBER                   Flatpak remote id
 --name VALUE                  Flatpak remote name to search by
 -h, --help                    Print help

[vagrant@centos7-hammer-devel hammer-cli-katello]$ hammer flatpak-remote delete --name "Fedora flatpak"
Could not delete the Flatpak Remote.:
  Error: Missing options to search organization.

[vagrant@centos7-hammer-devel hammer-cli-katello]$ hammer flatpak-remote delete --name "Fedora flatpak" --organization-id 1
Could not delete the Flatpak Remote.:
  Error: Unrecognised option '--organization-id'.

```

**Review by ianballou — COMMENTED** [review]
The help text for creating remotes is a little confusing, specifically for the URLs:

```
Usage:
    hammer flatpak-remote create [OPTIONS]

Options:
 --description VALUE                            Description of the flatpak remote
 --name VALUE                                   Name
 --organization[-id|-title|-label] VALUE/NUMBER Organization name/title/label/id
 --registry-url VALUE                           Optional Registry url. Will be updated when remote is scanned.
 --token VALUE                                  Token/password for the flatpak remote
 --url VALUE                                    Url
 --username VALUE                               Username for the flatpak remote
 -h, --help                                     Print help
```

It isn't quite clear what the difference is between the `url` and the `registry-url`. It also isn't clear why someone would need to enter a `registry-url`.

**Comment by ianballou** [comment]
Besides my comments above, the general workflows seem to be working well :+1:

File: lib/hammer_cli_katello/flatpak_remote.rb [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    class DeleteCommand < HammerCLIKatello::DeleteCommand
+      include OrganizationOptions
+      success_message _('Flatpak Remote deleted.')
+      failure_message _('Could not delete the Flatpak Remote.')
+
+      build_options
+    end
+
+    class ScanCommand < HammerCLIKatello::SingleResourceCommand
```
Comment (ianballou) [inline lib/hammer_cli_katello/flatpak_remote.rb:62]:
```suggestion
    class ScanCommand < HammerCLIKatello::SingleResourceCommand
      include OrganizationOptions
```

File: lib/hammer_cli_katello/flatpak_remote.rb [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    resource :flatpak_remotes
+
+    class ListCommand < HammerCLIKatello::ListCommand
+      include OrganizationOptions
+      output do
+        field :id, _('ID')
+        field :name, _('Name')
+        field :url, _('URL')
+        field :description, _('Description')
+        field :username, _('User')
```
Comment (ianballou) [inline lib/hammer_cli_katello/flatpak_remote.rb:13]:
```suggestion
        field :username, _('Username')
```

File: test/functional/flatpak_remote/list_test.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+  it "lists flatpak remotes and returns empty response" do
+    ex = api_expects(:flatpak_remotes, :index, 'flatpak list') do |par|
+      par['page'] == 1 && par['per_page'] == 1000
+    end
+
+    ex.returns(empty_response)
+
+    expected_result = success_result("---|------|-----|-------------|------|-------------
+ID | NAME | URL | DESCRIPTION | USER | REGISTRY URL
```
Comment (ianballou) [inline test/functional/flatpak_remote/list_test.rb:43]:
```suggestion
ID | NAME | URL | DESCRIPTION | USERNAME | REGISTRY URL
```

File: lib/hammer_cli_katello/flatpak_remote_repository.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+          field :digest, _("Manifest Digest")
+          field :tags, _("Manifest tags")
+        end
+      end
+      build_options do |o|
+        o.expand(:all).including(:flatpak_remotes, :organizations)
+      end
+    end
+
+    class MirrorCommand < HammerCLIKatello::SingleResourceCommand
```
Comment (ianballou) [inline lib/hammer_cli_katello/flatpak_remote_repository.rb:39]: Perhaps unrelated to this PR, but do you know what "organization title" is? 

We typically only have ID / Name / Label...

```
Usage:
    hammer flatpak-remote remote-repository mirror [OPTIONS]

Options:
 --async                                        Do not wait for the task
 --flatpak-remote[-id] VALUE/NUMBER             Name/Id of associated flatpak remote
 --id NUMBER                                    Flatpak remote repository id
 --name VALUE                                   Name to search by
 --organization[-id|-title|-label] VALUE/NUMBER Name/Title/Label/Id of associated organization
 --product[-id] VALUE/NUMBER                    Product name/id to mirror the remote repository to
 -h, --help                                     Print help
```

**Review by ianballou — COMMENTED** [review]
Some more comments:

File: lib/hammer_cli_katello/flatpak_remote_repository.rb (hunk 4 above)
Comment (sjha4) [reply lib/hammer_cli_katello/flatpak_remote_repository.rb:39]:
```
[2] pry(main)> org.title
=> "Default Organization"
[3] pry(main)> org.name
=> "Default Organization"
[4] pry(main)> org.label
=> "Default_Organization"
```

Title seems to be the same as name.

Comment (ianballou) [reply lib/hammer_cli_katello/flatpak_remote_repository.rb:39]: Interesting!

**Review by ianballou — APPROVED** [review]
Looks good!

**Review by vsedmik — APPROVED** [review]
Looks good to me.

**Comment by sjha4** [comment]
Merging.. 🎉

**Pull request merged**